account_invoice_margin/report/account_invoice_report.py

```diff
@@ -8,11 +8,13 @@ class AccountInvoiceReport(models.Model):
     _inherit = "account.invoice.report"
 
     margin = fields.Float(string='Margin', readonly=True)
+    margin_percent = fields.Float(string='Margin (%)', readonly=True)
 
     def _select(self):
         select_str = super()._select()
-        return '%s, sub.margin' % select_str
+        return '%s, sub.margin, sub.margin_percent' % select_str
 
     def _sub_select(self):
         select_str = super()._sub_select()
-        return '%s, SUM(ail.margin_signed) AS margin' % select_str
+        return '%s, SUM(ail.margin_signed) AS margin, ' \
+               'AVG(ai.margin_percent) AS margin_percent' % select_str
```

Comment:
I think it's wrong for margin_percent. The average of margin is not the margin of whole invoices. To test, 
make a invoice with the same customer.
- an invoice of 1000€, margin 20%
- a invoice of 2000€, margin 40%

with the current code, the margin will be 30%, that is wrong.

Reply:
Neither of both (sum or avg) has too much sense talking about percentage, but a group operator should be put. The only way to get the proper one is to inherit `read_group` and make the computation for each group, in a similar way I did this:

https://github.com/OCA/hr-attendance/blob/bcc7b04af14a52be994ebdc6ee9d52eab361303e/hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py#L239-L244

and it requires that the other measure is populated as well.

Reply:
Thank you @legalsylvain for your example. I have changed percentage computation from invoice to invoice line.

Reply:
@pedrobaeza awesome. Your example will help me a lot. Let me check something and work on more user test case.